### PR TITLE
swaynag: simplify several things

### PR DESCRIPTION
**Pull request by noocsharp**

### CODE REVIEW

**Comment by emersion** [comment]
`PATH_MAX` is a lie. Paths can be longer.

File: swaynag/swaynag.c [hunk 1]
```diff
@@ -63,12 +39,12 @@ static void swaynag_button_execute(struct swaynag *swaynag,
...
 				// Child of the child. Will be reparented to the init process
 				char *terminal = getenv("TERMINAL");
-				if (button->terminal && terminal && strlen(terminal)) {
+				if (button->terminal && terminal && *terminal) {
 					sway_log(SWAY_DEBUG, "Found $TERMINAL: %s", terminal);
-					if (!terminal_execute(terminal, button->action)) {
-						swaynag_destroy(swaynag);
-						_exit(EXIT_FAILURE);
-					}
+					execlp(terminal, terminal, "-e", "sh", "-c", button->action, NULL);
```
Comment (emersion) [inline swaynag/swaynag.c:44]: Some terminals use `-e cmd arg1 arg2 arg3`, some terminals use `-e "cmd arg1 arg2 arg3"`. The script allowed to handle both.

**Comment by noocsharp** [comment]
I just dropped the objectionable patches.

**Review by emersion — APPROVED** [review]
LGTM, thanks!

**Pull request merged**